From libc-ports-return-507-listarch-libc-ports=sources dot redhat dot com at sourceware dot org Thu Oct 19 20:30:39 2006
Return-Path: <libc-ports-return-507-listarch-libc-ports=sources dot redhat dot com at sourceware dot org>
Delivered-To: listarch-libc-ports at sources dot redhat dot com
Received: (qmail 25664 invoked by alias); 19 Oct 2006 20:30:38 -0000
Received: (qmail 25654 invoked by uid 22791); 19 Oct 2006 20:30:37 -0000
X-Spam-Status: No, hits=-2.2 required=5.0 	tests=AWL,BAYES_00,RCVD_ILLEGAL_IP,SPF_PASS
X-Spam-Check-By: sourceware.org
Received: from mail.codesourcery.com (HELO mail.codesourcery.com) (65.74.133.4)     by sourceware.org (qpsmtpd/0.31) with ESMTP; Thu, 19 Oct 2006 20:30:34 +0000
Received: (qmail 2800 invoked from network); 19 Oct 2006 20:30:32 -0000
Received: from unknown (HELO digraph.polyomino.org.uk) (joseph@127.0.0.2)   by mail.codesourcery.com with ESMTPA; 19 Oct 2006 20:30:32 -0000
Received: from jsm28 (helo=localhost) 	by digraph.polyomino.org.uk with local-esmtp (Exim 4.63) 	(envelope-from <joseph@codesourcery.com>) 	id 1GaeX9-0002bI-PF 	for libc-ports@sourceware.org; Thu, 19 Oct 2006 20:30:31 +0000
Date: Thu, 19 Oct 2006 20:30:31 +0000 (UTC)
From: "Joseph S dot  Myers" <joseph at codesourcery dot com>
X-X-Sender: jsm28@digraph.polyomino.org.uk
To: libc-ports at sourceware dot org
Subject: Update splice prototypes for ARM and MIPS
Message-ID: <Pine.LNX.4.64.0610192029570.8725@digraph.polyomino.org.uk>
MIME-Version: 1.0
Content-Type: TEXT/PLAIN; charset=US-ASCII
Mailing-List: contact libc-ports-help at sourceware dot org; run by ezmlm
Precedence: bulk
List-Subscribe: <mailto:libc-ports-subscribe at sourceware dot org>
List-Post: <mailto:libc-ports at sourceware dot org>
List-Help: <mailto:libc-ports-help at sourceware dot org>, <http://sourceware dot org/lists dot html#faqs>
Sender: libc-ports-owner at sourceware dot org
Delivered-To: mailing list libc-ports at sourceware dot org

The splice prototypes in ports need updating to correspond with those
used for non-ports architectures.

This patch updates ARM and MIPS.  Other targets also need either the
prototype updated (hppa) or the whole splice etc. section of the file
added (am33, cris, m68k).

2006-10-19  Joseph S. Myers  <joseph@codesourcery.com>

	* sysdeps/unix/sysv/linux/arm/bits/fcntl.h (splice): Update
	prototype.
	* sysdeps/unix/sysv/linux/mips/bits/fcntl.h (splice): Update
	prototype.

---
 sysdeps/unix/sysv/linux/arm/bits/fcntl.h  |    3 ++-
 sysdeps/unix/sysv/linux/mips/bits/fcntl.h |    3 ++-
 2 files changed, 4 insertions(+), 2 deletions(-)

Index: glibc-ports-2.5/sysdeps/unix/sysv/linux/arm/bits/fcntl.h
===================================================================
--- glibc-ports-2.5.orig/sysdeps/unix/sysv/linux/arm/bits/fcntl.h
+++ glibc-ports-2.5/sysdeps/unix/sysv/linux/arm/bits/fcntl.h
@@ -223,7 +223,8 @@ extern int vmsplice (int __fdout, const 
 		     unsigned int __flags);
 
 /* Splice two files together.  */
-extern int splice (int __fdin, int __fdout, size_t __len, unsigned int __flags)
+extern int splice (int __fdin, __off64_t *__offin, int __fdout,
+		   __off64_t *__offout, size_t __len, unsigned int __flags)
     __THROW;
 
 /* In-kernel implementation of tee for pipe buffers.  */
Index: glibc-ports-2.5/sysdeps/unix/sysv/linux/mips/bits/fcntl.h
===================================================================
--- glibc-ports-2.5.orig/sysdeps/unix/sysv/linux/mips/bits/fcntl.h
+++ glibc-ports-2.5/sysdeps/unix/sysv/linux/mips/bits/fcntl.h
@@ -237,7 +237,8 @@ extern int vmsplice (int __fdout, const 
 		     unsigned int __flags);
 
 /* Splice two files together.  */
-extern int splice (int __fdin, int __fdout, size_t __len, unsigned int __flags)
+extern int splice (int __fdin, __off64_t *__offin, int __fdout,
+		   __off64_t *__offout, size_t __len, unsigned int __flags)
     __THROW;
 
 /* In-kernel implementation of tee for pipe buffers.  */
